lib/matplotlib/ticker.py

```diff
@@ -902,14 +902,14 @@ def __call__(self, x, pos=None):
         fx = math.log(abs(x)) / math.log(b)
         isDecade = is_close_to_int(fx)
         exponent = np.round(fx) if isDecade else np.floor(fx)
-        coeff = np.round(x / b ** exponent)
+        coeff = np.round(abs(x) / b ** exponent)
         if coeff in self.sublabel:
             if not isDecade and self.labelOnlyBase:
-                s = ''
+                return ''
             elif x > 10000:
-                s = '%1.0e' % x
+                s = '%1.0e' % abs(x)
```

Comment:
I'll fix the abs stuff. Another solution is not to deal with the sign as it is done in the other formatter (or remove the explicit dealing with sign as it is done right now, I am not sure why)
Honestly, I think this whole part should be refactored and rewritten. I just wanted a quick and easy fix in this PR, as this is a critical bug for 2.0 and don't think we should refactor all this before the release.